iothub/service/src/ServiceAuthenticationWithSharedAccessPolicyKey.cs

```diff
@@ -21,59 +21,59 @@ public sealed class ServiceAuthenticationWithSharedAccessPolicyKey : IAuthentica
         /// <param name="key">Key associated with the shared access policy.</param>
         public ServiceAuthenticationWithSharedAccessPolicyKey(string policyName, string key)
         {
-            this.SetPolicyName(policyName);
-            this.SetKey(key);
+            SetPolicyName(policyName);
+            SetKey(key);
         }
 
         public string PolicyName
         {
-            get { return this.policyName; }
-            set { this.SetPolicyName(value);}
+            get => _policyName;
```

Comment:
this can be simplified to remove the backing field and have setter with null check

Reply:
How do you perform validation in a property without a backing field?

Reply:
You are right, auto properties cannot perform validations